### DISCUSSION TITLE
how to add L2 cache

### DISCUSSION THREAD

**Discussion by ghost** [mortbopet/Ripes#328] [Q&A]
hello,
how to modify the source code to simulation multi-level cache?

**Comment by Et-okapi** [comment]
same question......
I don't understand the architecture of Ripes simulator...

**Reply by ghost** [reply]
我也是要做bonus project可以來合作一下，雖然我不是交大的同學。

**Reply by austincortopassi** [reply]
Have either of you done any work on this in the past 10 months?

**Reply by mortbopet** [reply]
> Have either of you done any work on this in the past 10 months?

Not as far as I am aware of.

**Reply by nicholas9237** [reply]
想問學長姐最後有做出來嗎

**Reply by Et-okapi** [reply]
@nicholas9237 後來不想做了放假去，加油><

**Reply by changfish** [reply]
答案從這邊開始排隊嗎q

**Reply by lamblikecode** [reply]
在做 bonus project +1

**Reply by changfish** [reply]
有可以討論的地方嗎
目前有改一點點程式碼 但他Hit rate都沒變

**Reply by changfish** [reply]
欸 我發現是因為clock cycle超過10000所以值都沒變 現在好了

**Reply by nicholas9237** [reply]
要不要開個LINE群交流一下XD

**Reply by changfish** [reply]
我在想要怎麼給聯絡資訊 放在這邊有點恐怖

**Reply by lamblikecode** [reply]
我建一個匿名line群看看

**Reply by ddww0505** [reply]
請問要怎麼加:)

**Reply by fankai1oo** [reply]
同問，請問要怎麼加入呢?

**Comment by mortbopet** [comment]
A lot of the infrastructure for this has actually already been done, and is in Ripes already (work in this branch: https://github.com/mortbopet/Ripes/commit/c151bd7e6d1b81a9a62638210674883aa0376697).

 Specifically, you should be looking into the `CacheInterface` and the `m_nextLevelCache`-related variables. Some stuff is already in there, and some stuff is missing. Basically, what's needed are:
1. A way to add more cache levels in the GUI; one could imagine that multi-level caches are added via. new tabs in the gui (again, refer to the old branch since it already has some of that work https://github.com/mortbopet/Ripes/commits/n_caches/).
2. Once that is done, connecting that up to adding new `m_nextLevelCache`s, and propagating `CacheInterface::access` calls in-between caches whenever there is a cache miss.

Regardless, adding this change requires that you'll spend some time digging through and understanding the codebase, since it involves touching both GUI and simulator related things.

Hope that helps your further research!
